pkg/services/internal/whatsonchain/service.go

```diff
@@ -204,3 +242,25 @@ func (woc *WhatsOnChain) PostBEEF(ctx context.Context, beef *transaction.Beef, t
 
 	return &wdk.PostedBEEF{TxIDResults: txResults}, nil
 }
+
+// IsValidRootForHeight checks if the provided Merkle root is valid for the given block height.
+func (w *WhatsOnChain) IsValidRootForHeight(ctx context.Context, root *chainhash.Hash, height uint32) (bool, error) {
```

Comment:
```suggestion
func (woc *WhatsOnChain) IsValidRootForHeight(ctx context.Context, root *chainhash.Hash, height uint32) (bool, error) {
```

Please keep the receivers consistent.